TLSharp.Core/TelegramClient.cs

```diff
@@ -134,8 +136,9 @@ private async Task ReconnectToDcAsync(int dcId, CancellationToken token = defaul
             
             var dataCenter = new DataCenter (dcId, dc.IpAddress, dc.Port);
 
-            transport = new TcpTransport(dc.IpAddress, dc.Port, handler);
+            transport = new TcpTransport(dc.IpAddress, dc.Port, handler); // this will be gone after fix #937
             session.DataCenter = dataCenter;
+            session.Save(); // this is needed after fix #937
```

Comment:
if you don't save the session, the new tcpTransport created inside connectAsync (firx 937), will reload the session with the old datacenter value